vessim/cosim/computing_system.py

```diff
@@ -26,6 +26,15 @@ def init(self, sid, time_resolution, step_size, eid_prefix=None):
         self.step_size = step_size
         return super().init(sid, time_resolution, eid_prefix=eid_prefix)
 
+    def finalize(self) -> None:
+        """Stops http power meters' threads."""
+        super().finalize()
+        for model_instance in self.entities.values():
+            for power_meter in model_instance.power_meters: # type: ignore
+                if isinstance(power_meter, HttpPowerMeter):
+                    power_meter.update_thread.stop()
+                    power_meter.update_thread.join()
```

Comment:
```suggestion
                power_meter.finalize()
```

a clean way to do this without letting the `ComputingSystemSim` know about any implementation details of specific power meters, is to implement a method `finalize()` on the `PowerMeter` which is called in this function and, by default, does nothing. Currently, only the `HttpPowerMeter` overrides this method.